### ISSUE TITLE
New rule to require using pipefail on shell scripts using PIPE

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible-lint#442]
# Issue Type
- Feature request

# Desired Behaviour

Ansible lint should warning user that he needs to use `set -o pipefail` when he is using PIPE (`|`) in shells scripts as otherwise the command error code could be lost.
 
We have being succesfully using such extra test for a very long period of time and we would be happy to make it part of default ansible-lint rules: 

See https://github.com/openstack/tripleo-quickstart-extras/blob/master/ci-scripts/ansible_rules/ShellPipefail.py

Example:
```
- shell: |
    df | grep '/dev'
```
This code may fail to return a non zero exit code from `df` if grep will still have a match.

The safer way to write this code would be
```
- shell: |
    set -o pipefail
    df | grep '/dev'
```

**Comment by ssbarnea** [comment]
Interesting surprise to see that there is a two years old PR adding this at https://github.com/ansible/ansible-lint/pull/199 -- not sure how it compares with the rule we have but I don't care which one will be merged as long it will be, hopefully soon.

**Comment by ssbarnea** [comment]
@awcrosby Thanks! Can't wait to see the new release.

**Comment by gmr** [comment]
FWIW I think this assumes a bit about the shell being used. Dash (Debian Almquist Shell) does not support -o pipefail.
